File: Agentic-AI-Travel-Advisor/shared/Core/Entities/TravelPackage.cs
using TravelAdvisor.Core.Enums;

namespace TravelAdvisor.Core.Entities;

public class TravelPackage
{
    public int Id { get; set; }
    public string AgentId { get; set; } = string.Empty;
    public int DestinationId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal Price { get; set; }
    public int DurationDays { get; set; }
    public ApprovalStatus ApprovalStatus { get; set; } = ApprovalStatus.Pending;

    public ApplicationUser Agent { get; set; } = null!;
    public Destination Destination { get; set; } = null!;
    public ICollection<PackageActivity> Activities { get; set; } = [];
    public ICollection<Booking> Bookings { get; set; } = [];
}

File: Agentic-AI-Travel-Advisor/mobile-app/lib/models/travel_package.dart
import 'package_activity.dart';

class TravelPackage {
  final int id;
  final String title;
  final String? description;
  final double price;
  final int durationDays;
  final int destinationId;
  final String destinationName;
  final String destinationCountry;
  final int activityCount;
  final List<PackageActivity>? activities;

  const TravelPackage({
    required this.id,
    required this.title,
    this.description,
    required this.price,
    required this.durationDays,
    required this.destinationId,
    required this.destinationName,
    required this.destinationCountry,
    required this.activityCount,
    this.activities,
  });

  factory TravelPackage.fromJson(Map<String, dynamic> json) {
    return TravelPackage(
      id: json['id'] as int,
      title: json['title'] as String,
      description: json['description'] as String?,
      price: (json['price'] as num).toDouble(),
      durationDays: json['durationDays'] as int,
      destinationId: json['destinationId'] as int,
      destinationName: json['destinationName'] as String,
      destinationCountry: json['destinationCountry'] as String,
      activityCount: json['activityCount'] as int? ?? 0,
      activities: json['activities'] != null
          ? (json['activities'] as List)
              .map((a) => PackageActivity.fromJson(a))
              .toList()
          : null,
    );
  }
}
